### ISSUE TITLE
SyntaxError: invalid syntax

### ISSUE THREAD

**Issue by zqqw** [linux-audit/audit-userspace#7]
Enabling the option of Python3 bindings in Void Linux I got an error, although the build completed.
It seemed to originate from the audit Python src.
```
/audit-2.7.3$ ag "def set_aumessage_mode"
bindings/swig/python/audit.py
1782:def set_aumessage_mode(mode, debug):

bindings/swig/python3/audit.py
1626:def set_aumessage_mode(mode: 'message_t', debug: 'debug_message_t') -> "void":



 Warning in make output:
libtool: warning: remember to run 'libtool --finish /usr/lib/python3.5/site-packages'
 /usr/bin/mkdir -p '/destdir//audit-2.7.3/usr/lib/python3.5/site-packages'
 /builddir/.xbps-audit/wrappers/install -c -m 644 audit.py '/destdir//audit-2.7.3/usr/lib/python3.5/site-packages'
Byte-compiling python modules...
audit.py  File "/usr/lib/python3.5/site-packages/audit.py", line 719
    def set_aumessage_mode(mode: 'message_t', debug: 'debug_message_t') -> "void":
                               ^
SyntaxError: invalid syntax


Byte-compiling python modules (optimized versions) ...
audit.py
make[4]: Nothing to be done for 'install-data-am'.
make[4]: Leaving directory '/builddir/audit-2.7.3/bindings/swig/python3'
```
[audit273andpython3op.txt](https://github.com/linux-audit/audit-userspace/files/805018/audit273andpython3op.txt)

**Comment by stevegrubb** [comment]
This code is created by SWIG. Not sure what can be done about it since it pretty much does everything.

**Comment by pcmoore** [comment]
A quick skim of http://www.swig.org/Doc3.0/Python.html#Python_python3support makes me wonder if simply passing the '-py3' parameter to SWIG when building for Python3 would resolve this?

**Comment by zqqw** [comment]
```
audit-2.7.3$ ag SWIG_FLAGS
bindings/swig/python/Makefile.in
392:SWIG_FLAGS = -python
778:	swig -o audit_wrap.c ${SWIG_FLAGS} ${SWIG_INCLUDES} ${srcdir}/../src/auditswig.i 

bindings/swig/python/Makefile.am
25:SWIG_FLAGS = -python
37:	swig -o audit_wrap.c ${SWIG_FLAGS} ${SWIG_INCLUDES} ${srcdir}/../src/auditswig.i 

bindings/swig/python3/Makefile.in
392:SWIG_FLAGS = -python -py3 -modern
778:	swig -o audit_wrap.c ${SWIG_FLAGS} ${SWIG_INCLUDES} ${srcdir}/../src/auditswig.i 

bindings/swig/python3/Makefile.am
26:SWIG_FLAGS = -python -py3 -modern
38:	swig -o audit_wrap.c ${SWIG_FLAGS} ${SWIG_INCLUDES} ${srcdir}/../src/auditswig.i 
```
That did sound like a nice idea, but I think - probably - it uses that anyway from a quick search of the source code. I wonder if there is something about the little arrow under the colon pointing to the problem syntax, what it should say there though is another matter.
swig-3.0.12_1 is the version in Void, and it is the latest version. It isn't urgent that Python3 works incidentally, I was only trying to support the possibility of using it for Void users as Python2 is legacy now.

**Comment by zqqw** [comment]
"This code is created by SWIG. Not sure what can be done about it since it pretty much does everything."
Sorry, I hadn't quite understood the meaning, being unfamiliar with using SWIG.
So the problem code is automatically generated by your SWIG and Void's SWIG cannot then read it, making it a problem with Void's SWIG build options or a version incompatibility.
The answer is then to either fix Void SWIG, use it to regenerate the files, or feed it some option to make it use compatible parsing methods.
Thank you for the help.

**Comment by stevegrubb** [comment]
Checked the code with pylint. It seems like function annotation is the problem.
http://www.swig.org/Doc1.3/Python.html#Python_nn74
Since pyc files are in fact generated and semanage doesn't seem to crash, I'm guessing some build tool somewhere doesn't know about function annotation but the real compiler does.

**Comment by zqqw** [comment]
My current theory is it is due to me trying to build both Python and Python3 bindings at the same time:
https://lists.gnu.org/archive/html/automake/2014-11/msg00006.html
The syntax that fails seems to be valid Python3 but not Python. autoconf detects both types of Python but py-compile which is created by automake is using only python to byte compile the py files.  There are a few other bugs and things online that are similar. The link above has a link to a possible solution, by adding some things to the automake setup files:
https://www.technovelty.org/debian/cdbs-autotools-python.html
If this was the case it is an automake issue and not because of SWIG.
(Or just build one variant and forget the other.)
The build details are here if it was relevant: 
https://github.com/voidlinux/void-packages/pull/5858
Although that is just the first attempt. The python module failing isn't affecting the build, the issues there are just some fine tuning, enabling ARM support and excluding MUSL.

**Comment by stevegrubb** [comment]
Fixed by commit cc55d03. I guess we need to override the byte compile macro.
